keras_nlp/models/roberta.py

```diff
@@ -170,6 +171,184 @@ def get_config(self):
+     - Concatenate all input segments, adding a single `start_value` at the
+       start of the entire sequence, `[end_value, end_value]` at the end of
+       each segment save the last, and a single `end_value` at the end of the
+       entire sequence.
+     - Pad the resulting sequence to `sequence_length` using `pad_tokens`.
+
+    Input should be either a `tf.RaggedTensor` or a dense `tf.Tensor`, and
+    either rank-1 or rank-2.
+
+    Args:
+        sequence_length: The desired output length.
+        start_value: The id or token that is to be placed at the start of each
+            sequence (called `"<s>"` for RoBERTa). The dtype must match the dtype
+            of the input tensors to the layer.
+        end_value: The id or token that is to be placed at the end of each
+            input segment (called `"</s>"` for RoBERTa). The dtype much match the
+            dtype of the input tensors to the layer.
+        pad_value: The id or token that is to be placed into the unused
+            positions after the last segment in the sequence
+            (called "<pad>" for RoBERTa).
+        truncate: The algorithm to truncate a list of batched segments to fit a
+            per-example length limit. The value can be either `round_robin` or
+            `waterfall`:
+                - `"round_robin"`: Available space is assigned one token at a
+                    time in a round-robin fashion to the inputs that still need
+                    some, until the limit is reached.
+                - `"waterfall"`: The allocation of the budget is done using a
+                    "waterfall" algorithm that allocates quota in a
+                    left-to-right manner and fills up the buckets until we run
+                    out of budget. It support arbitrary number of segments.
+
+    Returns:
+        A dense, packed token sequence.
+    """
+
+    def __init__(
+        self,
+        sequence_length,
+        start_value,
+        end_value,
+        pad_value=None,
+        truncate="round_robin",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.sequence_length = sequence_length
+        if truncate not in ("round_robin", "waterfall"):
+            raise ValueError(
+                "Only 'round_robin' and 'waterfall' algorithms are "
+                "supported. Received %s" % truncate
+            )
+        self.truncate = truncate
+        self.start_value = start_value
+        self.end_value = end_value
+        self.pad_value = pad_value
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "sequence_length": self.sequence_length,
+                "start_value": self.start_value,
+                "end_value": self.end_value,
+                "pad_value": self.pad_value,
+                "truncate": self.truncate,
+            }
+        )
+        return config
+
+    def _sanitize_inputs(self, inputs):
```

Comment:
I don't think most of sanitize_inputs and convert_dense are not needed, as we control the inputs to this function from the tokenizer. We know we are going to have a rank 1 or 2 ragged input right?